Software/lib/GNSE_HAL/GNSE_hal.c

```diff
@@ -23,10 +23,74 @@
 #include "GNSE_hal.h"
 #include "GNSE_acc.h"
 #include "GNSE_bm.h"
+#include "GNSE_lpm.h"
 #include "GNSE_flash.h"
 #include "SHTC3.h"
 #include "BUZZER.h"
 
+/**
+  * @brief  Generic error handler
+  * @note   Weak implementation that can be overridden by the application layer
+  * @param  none
+  * @return none
+  */
+__weak void GNSE_Error_Handler(void)
+{
+    GNSE_BSP_LED_Init(LED_RED);
+    GNSE_BSP_LED_On(LED_RED);
+    GNSE_LPM_EnterLowPower();
+    while (1)
+    {
+    }
+}
+
+/**
+  * @brief  Initialises the clock peripherals
+  * @note   Weak implementation that can be overridden by the application layer
+  * @param  none
+  * @return CLK_op_result_t
+  */
+__weak CLK_op_result_t GNSE_CLK_Init(void)
+{
+    RCC_OscInitTypeDef RCC_OscInitStruct = {0};
+    RCC_ClkInitTypeDef RCC_ClkInitStruct = {0};
+
+    /** Configure LSE Drive Capability
```

Comment:
Would be great if we can improve the commenting in this function to match our [styling guide](https://discuss.thethingsindustries.com/t/c-c-embedded-code-guideline/103).
For example, I don't think we need two lines for all the comments in this function.

Reply:
Please apply to all comments within this function.